Polish (many small enhancements / fixes)

schnz opened the pull request:
Siehe einzelne commits. 

Weitere TODO's:
- [ ] Deadlines im Header anzeigen! (Oderirgendwo anders im layout. Ggf. farbig markieren, je nachdem ob abgelaufen oder nicht)
- [ ] Menüpunkte besser beschreiben, z.B. SHOW submissions, EDIT conference, ...
- [x] Reload state after changing deadlines
- [ ] Use permission permission-only="'AUTHOR'" instead of ng-show="u.isAuthor"
- [ ] /submission/list -> column width fix (col-md-10!)
- [ ] .table-responsive class does not work well with ui-select! 
- [ ]  .card-columns css class does not work in FF at all (simple use bootstrap column wrapping, add margin-bottom to the panels)
- [ ] remove console.log() calls!
- [ ] Badges im menü (anzahl d. unapproved submissions/reviews)
- [ ] Logout seems to lead (in some cases?) to requests of authentication required services (when redirected to login page)
- [ ] Add status labels of submission to chair submission overview

schnz committed:
Provide feedback for form submission when assigning reviewers to papers

schnz committed:
Fixed createUser.js script again. Why do we require lowercase usernames in the first place?

schnz committed:
AuthService.logout does now only delete local data and do not use the loopback method User.logout as it did not succeed in several cases

schnz committed:
app.js: .config before .run

schnz committed:
Added new link in navigation menu Conference Management > Conference (to let user edit deadlines etc.)

schnz committed:
Manage Conference: When updating a conference, also invalidate the cached conference model in ConferenceService and reload state in order to update header navigation accordingly (e.g. when deadlines have changed)

schnz committed:
Submissions > Show: Panels do now have a 2-column layout

schnz committed:
Reload state after joining a new conference -> Navigation gets updated accordingly

schnz committed:
Implemented delete profile. (Design similar to GitHub when deleting repository. The user must enter his username in order to delete his profile permanently)

schnz committed:
Chair can now approve submissions in the submission overview (Conference Management > Submissions)

schnz committed:
Author is now able to finalize submissions in the submission overview (Submissions > Show). This decision is permanent. Once finalized the submission cannot be changed anymore.

schnz committed:
Show submission status in submission overview (Submissiony > Show)

schnz commented:
Die TODOS oben für den Moment einfach mal ignorieren, die sind für mich damit ich die nicht vergesse.
Weitere Änderungen bringe ich über einen neuen PR ein.
@tarekauel  schau bitte mal drüber und wenns ok ist mergen, sonst feedback (siehe einzelne commits für die genauen änderungen, kanst dann in #43 die häckchen auch setzen)

tarekbecker commented:
Login Page: && shows user drop down! (session storage, cookies all removed)

```
angular.js:13920 TypeError: Cannot read property 'then' of null
    at Object.isSubmissionPhase (http://localhost:3000/js/app.js:282:31)
    at new <anonymous> (http://localhost:3000/js/controllers/header.js:17:23)
    at Object.instantiate (http://localhost:3000/vendor/angular/angular.js:4733:14)
    at $controller (http://localhost:3000/vendor/angular/angular.js:10369:28)
    at Object.<anonymous> (http://localhost:3000/vendor/angular-ui-router/release/angular-ui-router.js:4081:28)
    at http://localhost:3000/vendor/angular/angular.js:1247:18
    at invokeLinkFn (http://localhost:3000/vendor/angular/angular.js:9934:9)
    at nodeLinkFn (http://localhost:3000/vendor/angular/angular.js:9335:11)
    at compositeLinkFn (http://localhost:3000/vendor/angular/angular.js:8620:13)
    at publicLinkFn (http://localhost:3000/vendor/angular/angular.js:8500:30)
```
- Chair shows "unkown chair@..." in author drop down
- Submission cards (chair) in chrome misaligned, "approve slider" is way too small

schnz committed:
- Fixed permissions in header
- Fixed initialization of ConferenceService
- Fixed user before save hook to also handle updates correctly
- Fixed "approved" switch width in chrome

schnz commented:
- Login Page: && shows user drop down! (session storage, cookies all removed)
  - Ich weiß nicht für das das && steht aber der Header zeigt jetzt nicht mehr fälschlicherweise die ganze navbar wenn man ausgeloggt ist (lag an dem TypeError der jetzt auch gefixt ist)
- Chair shows "unkown chair@..." in author drop down
  - War durch den "before save" hook verursacht. Ist jetzt gefixt.
- Submission cards (chair) in chrome misaligned, "approve slider" is way too small
  - Lasse den switch jetzt unter dem Titel anzeigen. Funktioniert erstmal in beiden Browsern.

schnz commented:
Bzgl harmonize:
Das über lasse ich doch dir. Es gibt drei views mit column width 8:
- views\author\submissions.list.html  (hier gehört ne 10er hin denk ich)
- views\user\profile.html
- views\user\signup.html

Dann gibts noch welche mit 6 und welche mit 10.
